core/src/include/Xios.hpp

```diff
@@ -28,34 +28,89 @@ class Xios : public Configured<Xios> {
     Xios();
     ~Xios();
 
-    void finalize();
+    void close_context_definition();
     void context_finalize();
+    void finalize();
     bool isInitialized();
 
     void configure() override;
     void configureServer();
-    void configureCalendar();
 
+    /* Date and duration */
+    std::string convertXiosDatetimeToString(cxios_date datetime, bool isoFormat = true);
+    void printCXiosDate(cxios_date date);
+    void printCXiosDuration(cxios_duration duration);
+
+    /* Calendar */
     cxios_date getCalendarOrigin();
     cxios_date getCalendarStart();
-    xios::CDate getCurrentDate();
     cxios_duration getCalendarTimestep();
+    int getCalendarStep();
+    std::string getCurrentDate(bool isoFormat = true);
     void setCalendarOrigin(cxios_date origin);
     void setCalendarStart(cxios_date start);
     void setCalendarTimestep(cxios_duration timestep);
-
-    void getCalendarConfiguration();
-
-    std::string getCalendarDate(bool isoFormat = true);
-    int getCalendarStep();
-
     void updateCalendar(int stepNumber);
-    void write(const std::string fieldstr, double* data, const int ni, const int nj);
-
-    std::string convertXiosDatetimeToString(cxios_date datetime, bool isoFormat = true);
 
-    void printCXiosDate(cxios_date date);
-    void printCXiosDuration(cxios_duration duration);
+    /* Axis */
+    void createAxis(std::string axisId); // TODO
+    void setAxisSize(std::string axisId, int size); // TODO
+    void setAxisValues(std::string axisId, std::vector<double> values); // TODO
+    int getAxisSize(std::string axisId);
+    std::vector<double> getAxisValues(std::string axisId);
```

Comment:
Oh that was another thing I wanted to comment on here. I've wrapped C-style arrays of `double`s with `std::vector` because I read the following in the neXtSIM-DG README:
> * All array operations should be done using std::vectors - not C-style arrays
> * The use of C-style pointers, new, and delete is strongly discouraged